hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestFileUtil.java

```diff
@@ -294,7 +292,7 @@ public void testFullyDeleteSymlinks() throws IOException {
     ret = FileUtil.fullyDelete(linkDir);
     Assert.assertTrue(ret);
     Assert.assertFalse(linkDir.exists());
-    Assert.assertEquals(3, del.list().length);
+    Assert.assertEquals(3, Objects.requireNonNull(del.list()).length);
```

Comment:
i'd be tempted to move to AssertJ here, e.g 

```
Assertions.assertThat(del.list)
  .describedAs("del list")
  .isNotNulll()
  .hasLength(3)
```

actually, given how many places  it us used, I'd factor this out into its own assertion used
through all the tests